vmdb/app/models/vm_scan.rb

```diff
@@ -474,13 +488,17 @@ def call_abort_retry(*args)
 
   def process_abort(*args)
     begin
+      vm = VmOrTemplate.find_by_id(self.target_id)
       unless self.context[:snapshot_mor].nil?
         mor = self.context[:snapshot_mor]
         self.context[:snapshot_mor] = nil
         set_status("Deleting snapshot before aborting job")
-        delete_snapshot(mor)
+        if vm.kind_of?(VmOpenstack)
+          vm.ext_management_system.vm_delete_evm_snapshot(vm, mor)
+        else
+          delete_snapshot(mor)
+        end
```

Comment:
I feel like this logic belongs in the delete_snapshot method, instead of out here.